src/Arcus.Security.Providers.DockerSecrets/DockerSecretsSecretProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration.KeyPerFile;
+using System;
+using System.Threading.Tasks;
+
+namespace Arcus.Security.Providers.DockerSecrets
+{
+    /// <summary>
+    /// Represents an <see cref="ISecretProvider" /> that provides access to the Docker secrets mounted into the Docker container as files.
+    /// </summary>
+    public class DockerSecretsSecretProvider : ISecretProvider
+    {
+        private readonly KeyPerFileConfigurationProvider _provider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DockerSecretsSecretProvider"/> class.
+        /// </summary>
+        /// <param name="configurationSource">The configuration source that provides the Docker secrets mounted as files in the container.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="configurationSource"/> is <c>null</c></exception>
+        public DockerSecretsSecretProvider(KeyPerFileConfigurationSource configurationSource)
+        {
+            Guard.NotNull(configurationSource, nameof(configurationSource));
+            _provider = new KeyPerFileConfigurationProvider(configurationSource);
+            _provider.Load();
```

Comment:
So, it's OK to do it here, but we should make sure that it's the same approach as the UserSecrets. So either change it here or change it in the UserSecrets. That would be best.

Reply:
I've changed this even more.
In my opinion, the only thing that users of the `DockerSecretProvider` should be aware of, is the location where the secrets are stored inside the container.  How those secrets are being retrieved, is an implementation detail and users of `DockerSecretProvider`shouldn't be bothered with that.

Therefore, I've changed the constructor so that we only need to pass in the directory where the secrets are located.  The fact that we're now using the `KeyPerFileProvider` is an implementation detail and should be hidden from the callers.   If we would change the implementation of the `DockerSecretProvider` (f.i. no longer rely on the `KeyPerFileConfigProvider`), that should be transparent for the users of the `DockerSecretProvider`, and they should not be confronted with having to perform code changes for that.

Reply:
Ah, yes, I can agree on that. Can you create an issue for this to change it in the UserSecrets provider? Thanks. I'll resolve this.